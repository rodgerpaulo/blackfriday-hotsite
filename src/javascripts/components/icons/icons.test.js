import React from 'react';
import CloseIcon from './close-icon';
import BaseSvg from './base-svg';
import DiscountBadgeIcon from './discount-badge-icon';
import PercentBadgeIcon from './percent-badge-icon';
import ShoppingCartIcon from './shopping-cart-icon';
import renderer from 'react-test-renderer';

describe('Render Icons', () => {
  it('should render BaseSvg without crash', () => {
    const component = renderer.create(
      <BaseSvg width="24" height="24" color="#dddddd" />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render CloseIcon without crash', () => {
    const component = renderer.create(<CloseIcon />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render DiscountBadgeIcon without crash', () => {
    const component = renderer.create(<DiscountBadgeIcon />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render PercentBadgeIcon without crash', () => {
    const component = renderer.create(<PercentBadgeIcon />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render ShoppingCartIcon without crash', () => {
    const component = renderer.create(<ShoppingCartIcon />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
