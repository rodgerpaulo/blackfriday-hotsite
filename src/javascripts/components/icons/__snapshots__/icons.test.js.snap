// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Icons should render BaseSvg without crash 1`] = `
<svg
  className=""
  color="#dddddd"
  fill="#dddddd"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
/>
`;

exports[`Render Icons should render CloseIcon without crash 1`] = `
<svg
  className=""
  color="#505050"
  fill="#505050"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="none"
    fillRule="evenodd"
    opacity=".54"
  >
    <g
      fill="#505050"
      transform="translate(-415 -99)"
    >
      <g
        transform="translate(416 99)"
      >
        <polygon
          points="19 6.4 17.6 5 12 10.6 6.4 5 5 6.4 10.6 12 5 17.6 6.4 19 12 13.4 17.6 19 19 17.6 13.4 12"
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`Render Icons should render DiscountBadgeIcon without crash 1`] = `
<svg
  className=""
  color="#ffc62d"
  fill="#ffc62d"
  height="90"
  viewBox="0 0 78 90"
  width="78"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <polygon
      fill="#FFC62D"
      points="38.995 89.976 0 67.482 0 22.494 38.995 0 77.991 22.494 77.991 67.482"
    />
    <text
      fill="#262626"
      fontFamily="Arial-Black, Arial"
      fontSize="33"
      fontWeight="700"
      line-spacing="33"
    >
      <tspan
        x="16"
        y="50"
      >
        50
      </tspan>
    </text>
    <text
      fill="#262626"
      fontFamily="Arial-Black, Arial"
      fontSize="12"
      fontWeight="700"
    >
      <tspan
        x="59"
        y="39"
      >
        %
      </tspan>
    </text>
    <text
      fill="#262626"
      fontFamily="Arial-BoldMT, Arial"
      fontSize="9"
      fontWeight="bold"
    >
      <tspan
        x="15"
        y="62"
      >
        + BARATO
      </tspan>
    </text>
  </g>
</svg>
`;

exports[`Render Icons should render PercentBadgeIcon without crash 1`] = `
<svg
  className=""
  color="#ffc62d"
  fill="#ffc62d"
  height="90"
  viewBox="0 0 78 90"
  width="78"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <polygon
      fill="#dccca2"
      points="38.995 89.976 0 67.482 0 22.494 38.995 0 77.991 22.494 77.991 67.482"
    />
    <path
      d="m45.316 30h4.1211l-16.543 30.586h-4.082l16.504-30.586zm-24.316 7.5195c0-1.3281 0.28971-2.5911 0.86914-3.7891 0.57943-1.1979 1.4648-2.1191 2.6562-2.7637 1.1914-0.64453 2.6009-0.9668 4.2285-0.9668 2.3958 0 4.2838 0.68359 5.6641 2.0508 1.3802 1.3672 2.0703 3.1901 2.0703 5.4688 0 2.2917-0.6901 4.1211-2.0703 5.4883-1.3802 1.3672-3.2682 2.0508-5.6641 2.0508-1.6276 0-3.0371-0.32226-4.2285-0.9668-1.1914-0.64453-2.0768-1.5657-2.6562-2.7637s-0.86914-2.4674-0.86914-3.8086zm5.4883 0.039063c0 1.6146 0.26041 2.7865 0.78125 3.5156 0.33854 0.48177 0.83333 0.72266 1.4844 0.72266 0.65104 0 1.1393-0.24088 1.4648-0.72266 0.52084-0.72917 0.78125-1.901 0.78125-3.5156 0-1.6016-0.26041-2.7734-0.78125-3.5156-0.32552-0.46875-0.8138-0.70312-1.4648-0.70312-0.65104 0-1.1458 0.23437-1.4844 0.70312-0.52084 0.72917-0.78125 1.901-0.78125 3.5156zm15.156 15.449c0-1.3412 0.28646-2.6074 0.85938-3.7988 0.57292-1.1914 1.4583-2.1126 2.6562-2.7637 1.1979-0.65104 2.6042-0.97656 4.2188-0.97656 2.4089 0 4.3034 0.68684 5.6836 2.0605s2.0703 3.1999 2.0703 5.4785-0.6901 4.1048-2.0703 5.4785-3.2747 2.0605-5.6836 2.0605c-1.6146 0-3.0208-0.32552-4.2188-0.97656-1.1979-0.65104-2.0833-1.5723-2.6562-2.7637-0.57292-1.1914-0.85938-2.4577-0.85938-3.7988zm5.4883 0.039062c0 1.6146 0.26041 2.7865 0.78125 3.5156 0.32552 0.46875 0.8138 0.70312 1.4648 0.70312 0.65104 0 1.1458-0.23437 1.4844-0.70312 0.52084-0.72917 0.78125-1.901 0.78125-3.5156s-0.26041-2.793-0.78125-3.5352c-0.33854-0.46875-0.83333-0.70312-1.4844-0.70312-0.65104 0-1.1393 0.24088-1.4648 0.72266-0.52084 0.72917-0.78125 1.901-0.78125 3.5156z"
      fill="#ab9fb5"
    />
  </g>
</svg>
`;

exports[`Render Icons should render ShoppingCartIcon without crash 1`] = `
<svg
  className=""
  color="#fff"
  fill="#fff"
  height="36"
  viewBox="0 0 24 24"
  viewboxheight="24"
  viewboxwidth="24"
  width="36"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <path
      d="m7 18c-1.1 0-1.99 0.9-1.99 2s0.89 2 1.99 2 2-0.9 2-2-0.9-2-2-2zm-6-16v2h2l3.6 7.59-1.35 2.45c-0.16 0.28-0.25 0.61-0.25 0.96 0 1.1 0.9 2 2 2h12v-2h-11.58c-0.14 0-0.25-0.11-0.25-0.25l0.03-0.12 0.9-1.63h7.45c0.75 0 1.41-0.41 1.75-1.03l3.58-6.49c0.08-0.14 0.12-0.31 0.12-0.48 0-0.55-0.45-1-1-1h-14.79l-0.94-2h-3.27zm16 16c-1.1 0-1.99 0.9-1.99 2s0.89 2 1.99 2 2-0.9 2-2-0.9-2-2-2z"
      fill="#fff"
      fillRule="nonzero"
    />
  </g>
</svg>
`;
